shroud/wrapf.py

```diff
@@ -1510,6 +1512,8 @@ def wrap_function_impl(self, cls, node):
                     "procedure :: %s => %s"
                     % (fmt_func.F_name_function, fmt_func.F_name_impl)
                 )
+            if node.cpp_if:
+                self.type_bound_part.append("#endif")
```

Comment:
It might be helpful to add a comment to the generated code to indicate which `#if` this is ending.
E.g. 
```python
....append("#endif /* {} */ ".format ( node.cpp_if ) )
```

Reply:
(if fortran's  preprocessor supports comments)

Reply:
I've had portability issues with comments in c-preprocessor statements in Fortran before. So it's probably better to leave them off for now. Ideally, the amount of code that is controlled by the `ifdef` is small enough to fit on one screen.

Reply:
What about adding a comment on the previous/following line?